src/UnigramWordPredictor.java

```diff
@@ -51,7 +54,30 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
   public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
-    // TODO: Convert the trainingWords into neighborMap here
+    // iterate through trainingWords list (minus 1, because there is nothing to learn from last word)
+    for (int i = 0; i < trainingWords.size() - 1; i++) {
+
+      // store word at index, and next word
+      String word = trainingWords.get(i);
+      String nextWord = trainingWords.get(i+1);
+
+      // IF current word is NOT in neighborMap, create a List<String> "nextList" with containing the next value, then add (word, next) to neighborMap
+      if (!neighborMap.containsKey(word)) {
+
+        // create a list containing the next word
+        List<String> nextList = new ArrayList<String>();
+        nextList.add(nextWord);
+        
+        // add word and list to neigborMap
+        neighborMap.put(word, nextList);
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?